metriken-exposition/src/util.rs

```diff
@@ -0,0 +1,37 @@
+use std::fs::File;
+use std::io::{BufRead, BufReader, Seek};
+use std::path::Path;
+
+use parquet::errors::ParquetError;
+
+use crate::snapshot::Snapshot;
+use crate::{ParquetOptions, ParquetSchema};
+
+/// Converts a file with metrics in msgpack format to a parquet file.
+/// If successful, returns the number of rows written out to the parquet file.
+pub fn msgpack_to_parquet<P: AsRef<Path>>(input: P, output: P) -> Result<i64, ParquetError> {
```

Comment:
This restricts both parameters to be the same type, which is not what you want.
```suggestion
pub fn msgpack_to_parquet(input: impl AsRef<Path>, output: impl AsRef<Path>) -> Result<i64, ParquetError> {
```